provisioning/WFLY-21122-WildFly-Feature_pack-Family.adoc

```diff
@@ -0,0 +1,176 @@
+
+== Requirements
+
+The following feature-packs are part of the `wildfly` family:
+
+* WildFly EE feature-pack
+* WildFly feature-pack
+* WildFly Preview feature-pack
+
+A possible subset of criteria to bootstrap the approach for WildFly 39:
+
+* `jakarta-ee-deployment`: ability to deploy applications "a la" Jakarta EE.
+* `jakarta-ee`: An implementation of the Jakarta EE spec
+* `jakarta-ee-11`: An implementation of the Jakarta EE version 11
+* `microprofile`: An implementation of the MicroProfile spec
+* `microprofile-7.1`: An implementation of the MicroProfile spec version 7.1.
+
+The set of criteria can evolve (new ones added, some removed) during the life time of the WildFly project.
+
+Some criteria could have a longer life (e.g: `jakarta-ee`) than some others (e.g.: `microprofile-7.1` that is bound to a specific version).
+
+The benefit of criteria is that feature-packs that depend on WildFly feature-packs criteria will have a clearer contract 
+than the today feature-pack producer name (that doesn't capture the criteria information).
+For example, a feature-pack expecting `jakarta-ee-10` and not compatible with `jakarta-ee-12` 
+would be expected to fail when provisioned with a future WildFly release.
+
+Each feature-pack would expose the following set of criteria:
+
+* WildFly EE feature-pack: `jakarta-ee-deployment`, `jakarta-ee`, `jakarta-ee-11`
+* WildFly feature-pack: `microprofile`, `microprofile-7.1` + the following ones inherited from WildFly EE : `jakarta-ee-deployment`, `jakarta-ee`
+* WildFly Preview feature-pack: `jakarta-ee-deployment`, `jakarta-ee`, `jakarta-ee-11`, `microprofile`, `microprofile-7.1`
+
+=== Hard Requirements
+
+* Allow to provision WildFly or WildFly Preview with other feature-packs compatible with the defined criteria.
+* Suppress (or at least deprecate and remove in a next version) the extra feature-packs that have been defined to work with WildFly Preview: cloud, datasources and GRPC.
+* No enforcement for extra feature-packs to benefit from this feature. Already released feature-pack could be provisioned with new WildFly feature-packs
+
+NOTE: That is outside of this proposal but WildFly and WildFly Preview feature-packs, 
+in order to be valid alternatives, must expose a common set of Galleon `features`, `packages` and `layers`.
+
+=== Implementation notes
+
+The changes in the WildFly project are:
+
+* Bump to Galleon 7.x and WildFly Galleon Plugins 8.1.x
+* Update the three `wildfly-feature-pack-build.xml` files to define their family and criteria.
+* Update the WildFly full `wildfly-feature-pack-build.xml` file to inherit criteria from its WildFly EE dependency.
+
+=== Non-Requirements
+
+* It is not a requirement that the same set of exposed criteria is present for each new WildFly Major releases.
+
+=== Future Work
+
+None
+
+== Backwards Compatibility
+
+* This evolution is transparent and doesn't require an evolution of the provisioning tooling (WildFly Maven Plugin, WildFly Glow, ...).
```

Comment:
in the presentation you did in the community call, wasn't there new Glow CLI options to specify which variants you wanted to provision?

Reply:
Good point. This proposal doesn't require changes in WildFly Glow. WildFly Glow has already a support  for WildFly Preview. Nevertheless I have added the impact on WildFly Glow to allow for future WildFly variants.